### PR TITLE
Add experimental packaging separation of pipeline authoring SDK.

### PR DESCRIPTION
**Pull request by charlesccychen**

### CODE REVIEW

**Comment by charlesccychen** [comment]
CC: @neuromage @zhitaoli @ruoyu90 @1025KB

**Review by zhitaoli — COMMENTED** [review]
Can you write some blurbs in contributing.md to explain how these packages relate to each other?

File: setup.py [hunk 1]
```diff
@@ -164,9 +169,34 @@ def run(self):
...
+  # Monolithic development package with the entirety of `tfx.*` and the full
+  # set of dependencies. Functionally equivalent to the union of the "tfx" and
+  # "tfx-pipeline-sdk" packages.
+  install_requires = dependencies.make_required_install_packages()
+  packages = find_namespace_packages(include=['tfx.*'])
+elif package_config.PACKAGE_NAME == 'tfx-pipelines-sdk':
+  # Core framework-neutral TFX pipeline SDK, without dependency on component-
+  # specific packages like "tensorflow" and "apache-beam".
+  install_requires = dependencies.make_pipeline_sdk_required_install_packages()
+  packages = find_namespace_packages(include=['tfx.dsl', 'tfx.proto'])
```
Comment (zhitaoli) [inline setup.py:185]: Should this be an identiccal list with exclude on line 194 below? If so, move it to a local variable?

File: setup.py [hunk 2]
```diff
@@ -164,9 +169,34 @@ def run(self):
 with open('README.md') as fp:
   _LONG_DESCRIPTION = fp.read()
 
+package_name = package_config.PACKAGE_NAME
+install_requires = dependencies.make_required_install_packages()
+
```
Comment (zhitaoli) [inline setup.py:174]: Consider log/print the PACKAGE_NAME selected here?

File: setup.py [hunk 3]
```diff
@@ -37,6 +38,10 @@
 from tfx.tools import resolve_deps
 from wheel import bdist_wheel
 
+# Prefer to import `package_config` from the setup.py script's directory.
+sys.path.insert(0, os.path.dirname(__file__))
+import package_config
```
Comment (zhitaoli) [inline setup.py:43]: Add some explanation on how this selected during wheel building?

Comment (charlesccychen) [reply setup.py:43]: Done

File: setup.py (hunk 2 above)
Comment (charlesccychen) [reply setup.py:185]: Done.

File: setup.py (hunk 3 above)
Comment (charlesccychen) [reply setup.py]: Done.

File: setup.py (hunk 1 above)
Comment (charlesccychen) [reply setup.py]: Done.

File: package_config.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Configuration for the "tfx-dev" package.
+
+Monolithic development package with the entirety of `tfx.*` and the full
+set of dependencies. Functionally equivalent to the union of the "tfx" and
+"tfx-pipeline-sdk" packages.
```
[the remark this answers is not in the document]
Comment (charlesccychen) [reply package_config.py:19]: Added explanation in `package_build/README.md`.

**Pull request closed without merging**